api/s2n.h

```diff
@@ -136,16 +136,48 @@ extern int *s2n_errno_location(void);
  *
  * This enum is optimized for use in C switch statements. Each value in the enum represents
  * an error "category".
+ *
+ * s2n-tls organizes errors into different "types" to allow applications to do logic on error
+ * values without catching all possibilities. Applications using non-blocking I/O should check
+ * error type to determine if the I/O operation failed because it would block or for some other
+ * error. To retrieve the type for a given error use `s2n_error_get_type()`. Applications should
+ * perform any error handling logic using these high level types:
+ *
+ * Here's an example that handles errors based on type:
+ *
+ * ```c
+ * s2n_errno = S2N_ERR_T_OK;
+ * if (s2n_recv(conn, &blocked) < 0) {
+ *     switch(s2n_error_get_type(s2n_errno)) {
+ *         case S2N_ERR_T_BLOCKED:
+ *             return -1;
+ *         case S2N_ERR_T_CLOSED:
+ *             return 0;
+ *         case S2N_ERR_T_IO:
+ *             handle_io_err();
+ *             return -1;
+ *         case S2N_ERR_T_PROTO:
+ *             handle_proto_err();
+ *             return -1;
+ *         case S2N_ERR_T_ALERT:
+ *             log_alert(s2n_connection_get_alert(conn));
+ *             return -1;
+ *         default:
+ *             log_other_error();
+ *             return -1;
+ *     }
+ * }
+ * ```
  */
 typedef enum {
-    S2N_ERR_T_OK=0,
-    S2N_ERR_T_IO,
-    S2N_ERR_T_CLOSED,
-    S2N_ERR_T_BLOCKED,
-    S2N_ERR_T_ALERT,
-    S2N_ERR_T_PROTO,
-    S2N_ERR_T_INTERNAL,
-    S2N_ERR_T_USAGE
+  S2N_ERR_T_OK=0, /** No error */
+  S2N_ERR_T_IO, /** Underlying I/O operation failed, check system errno */
```

Comment:
Generally we put comments above what they're commenting on, not inline. Does Doxygen require enum comments be inline?
```suggestion
  /* No error */
  S2N_ERR_T_OK=0,
  /* Underlying I/O operation failed, check system errno */
  S2N_ERR_T_IO, 
```

Reply:
Will be in next commit. Doxygen worked without inline